pkg/controller/server_status_request_controller.go

```diff
@@ -17,104 +17,100 @@ limitations under the License.
+	Ctx          context.Context
+	ServerStatus velero.ServerStatus
 
-	return c
+	Log logrus.FieldLogger
 }
 
-func (c *statusRequestController) processItem(key string) error {
-	log := c.logger.WithField("key", key)
-
-	log.Debug("Running processItem")
-	ns, name, err := cache.SplitMetaNamespaceKey(key)
-	if err != nil {
-		return errors.Wrap(err, "error splitting queue key")
-	}
+// +kubebuilder:rbac:groups=velero.io,resources=serverstatusrequests,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=velero.io,resources=serverstatusrequests/status,verbs=get;update;patch
+func (r *ServerStatusRequestReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithFields(logrus.Fields{
+		"controller":          "serverstatusrequest",
+		"serverStatusRequest": req.NamespacedName,
+	})
 
+	// Fetch the ServerStatusRequest instance.
 	log.Debug("Getting ServerStatusRequest")
-	req, err := c.lister.ServerStatusRequests(ns).Get(name)
-	// server status request no longer exists
-	if apierrors.IsNotFound(err) {
-		log.WithError(err).Debug("ServerStatusRequest not found")
-		return nil
+	statusRequest := &velerov1api.ServerStatusRequest{}
+	if err := r.Client.Get(r.Ctx, req.NamespacedName, statusRequest); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.WithError(err).Debug("ServerStatusRequest not found")
+			return ctrl.Result{}, nil
+		}
+
+		log.WithError(err).Debug("Error getting ServerStatusRequest")
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
 	}
-	if err != nil {
-		return errors.Wrap(err, "error getting ServerStatusRequest")
+
+	log = r.Log.WithFields(logrus.Fields{
+		"controller":          "serverstatusrequest",
+		"serverStatusRequest": req.NamespacedName,
+		"phase":               statusRequest.Status.Phase,
+	})
+
+	switch statusRequest.Status.Phase {
+	case "", velerov1api.ServerStatusRequestPhaseNew:
+		log.Info("Processing new ServerStatusRequest")
+
+		if err := r.ServerStatus.PatchStatusProcessed(r.Client, statusRequest, r.Ctx); err != nil {
+			log.WithError(err).Error("Unable to update the request")
+			return ctrl.Result{RequeueAfter: statusRequestResyncPeriod}, err
+		}
+	case velerov1api.ServerStatusRequestPhaseProcessed:
+		log.Debug("Checking whether ServerStatusRequest has expired")
+		expiration := statusRequest.Status.ProcessedTimestamp.Add(ttl)
+		if expiration.After(r.ServerStatus.Clock.Now()) {
+			log.Debug("ServerStatusRequest has not expired")
+			return ctrl.Result{RequeueAfter: statusRequestResyncPeriod}, nil
+		}
+
+		log.Debug("ServerStatusRequest has expired, deleting it")
+		if err := r.Client.Delete(r.Ctx, statusRequest); err != nil {
+			log.WithError(err).Error("Unable to delete the request")
+			return ctrl.Result{}, nil
+		}
+	default:
+		log.Errorf("unexpected ServerStatusRequest phase %q", statusRequest.Status.Phase)
+		return ctrl.Result{}, errors.Errorf("unexpected ServerStatusRequest phase %q", statusRequest.Status.Phase)
```

Comment:
We should either log the error or return it - seems like in this case we want to return it since somehow the `Phase` got set to something completely unexpected.